odoo/addons/base/models/qweb.py

```diff
@@ -655,115 +332,211 @@ def _compile_node(self, el, options):
-                iter=ast.Call(
-                    func=ast.Attribute(
-                        value=ast.Name(id='t_attrs', ctx=ast.Load()),
-                        attr='items',
-                        ctx=ast.Load()
-                        ),
-                    args=[], keywords=[],
-                    starargs=None, kwargs=None
-                ),
-                body=[ast.If(
-                    test=ast.BoolOp(
-                        op=ast.Or(),
-                        values=[
-                            ast.Name(id='value', ctx=ast.Load()),
-                            ast.Call(
-                                func=ast.Name(id='isinstance', ctx=ast.Load()),
-                                args=[
-                                    ast.Name(id='value', ctx=ast.Load()),
-                                    ast.Name(id='str', ctx=ast.Load())
-                                ],
-                                keywords=[],
-                                starargs=None, kwargs=None
-                            )
-                        ]
-                    ),
-                    body=[
-                        self._append(ast.Str(u' ')),
-                        self._append(ast.Name(id='name', ctx=ast.Load())),
-                        self._append(ast.Str(u'="')),
-                        self._append(ast.Call(
-                            func=ast.Name(id='escape', ctx=ast.Load()),
-                            args=[ast.Call(
-                                func=ast.Name(id='to_text', ctx=ast.Load()),
-                                args=[ast.Name(id='value', ctx=ast.Load())], keywords=[],
-                                starargs=None, kwargs=None
-                            )], keywords=[],
-                            starargs=None, kwargs=None
-                        )),
-                        self._append(ast.Str(u'"')),
-                    ],
-                    orelse=[]
-                )],
-                orelse=[]
-            )
-        ]
+    def _compile_options(self, el, varname, options, indent):
+        """
+        compile t-options and add to the dict the t-options-xxx values
+        """
+        t_options = el.attrib.pop('t-options', None)
+        directive = ''
+        if t_options:
+            directive = f't-options="{t_options}"'
+
+        code = []
+        dict_arg = []
+        for key in list(el.attrib):
+            if key.startswith('t-options-'):
+                value = el.attrib.pop(key)
+                directive += f' {key}="{value}"'
+                option_name = key[10:]
+                dict_arg.append(f'"{self._compile_str(option_name)}":{self._compile_expr(value)}')
+
+        if t_options and dict_arg:
+            code.append(self._indent(f"{varname} = {{**dict({self._compile_expr(t_options)}), {', '.join(dict_arg)}}}", indent))
```

Comment:
`t-options` ought be a dict literal, why are we feeding it to a `dict()`?

Reply:
It is not necessarily a literal dict. Nothing prevents having any python expression. But indeed whatever the method, dict, literal dict, the t-options should be a dict in the end. I will remove the dict